src/material/material.c

```diff
@@ -92,17 +94,19 @@ static int	surface_deep_copy(t_surface *dst, t_surface *src)
 
 static int	ppm_p6_deep_copy(t_ppm_p6 *dst, t_ppm_p6 *src)
 {
+	const size_t	size_in_bytes = sizeof(*src->pixels) * src->size;
+
 	dst->width = src->width;
 	dst->height = src->height;
 	dst->size = src->size;
 	if (src->pixels == NULL)
 		dst->pixels = NULL;
 	else
 	{
-		dst->pixels = malloc(sizeof(*src->pixels) * src->size);
+		dst->pixels = malloc(size_in_bytes);
 		if (dst->pixels == NULL)
 			return (-1);
-		ft_memcpy(dst->pixels, src->pixels, src->size);
```

Comment:
This did not copy the whole array as src->size does not represent the size of the array in bytes but in the number of elements.